pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/oom/metrics/xml/CycloTest.xml

```diff
@@ -0,0 +1,195 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<test-data>
+
+    <code-fragment id="full-example">
+        <![CDATA[
+        public class Complicated {
+          public void example() {
+            int x = 0, y = 1, z = 2, t = 2;
+            boolean a = false, b = true, c = false, d = true;
+
+            if (a && b || b && d) {
+              if (y == z) {
+                x = 2;
+              } else if (y == t && !d) {
+                x = 2;
+              } else {
+                x = 2;
+              }
+            } else if (c && d) {
+              while (z < y) {
+                x = 2;
+              }
+            } else if (a && !b) {
+              for (int n = 0; n < t; n++) {
+                x = 2;
+              }
+            } else {
+              switch (x) {
+              case 1:
+                x = 2;
+                break;
+
+              case 2:
+              case 3:
+                x = 2;
+                break;
+
+              default:
+                x = 2;
+                break;
+              }
+            }
+          }
+        }
+        ]]>
+    </code-fragment>
+
+    <test-code>
```

Comment:
I'm just remembering - there is also the (undocumented) feature of the testframework to re-initialize the rule before executing the test, e.g.

    <test-code reinitializeRule="true">

However, looking at the [code](https://github.com/pmd/pmd/blob/master/pmd-test/src/main/java/net/sourceforge/pmd/testframework/RuleTst.java#L304), this seems to be enabled by default anyway....
It was an idea, that you could trigger the "MetricsForceHook" that way...